src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/plugins/ConnectionPluginManager.java

```diff
@@ -47,18 +49,23 @@
   protected PropertySet propertySet = null;
   protected IConnectionPlugin headPlugin = null;
   ClusterAwareConnectionProxy proxy;
-  private final Thread shutdownHook = new Thread(this::releaseResources);
+  protected static final Queue<ConnectionPluginManager> instances;
+
+  static {
+    instances = new ConcurrentLinkedQueue<>();
+    Runtime.getRuntime().addShutdownHook(new Thread(ConnectionPluginManager::releaseAllResources));
+  }
 
   public ConnectionPluginManager(Log logger) {
     if (logger == null) {
```

Comment:
To be honest I am not sure if we want to throw a RuntimeException if a logger is not provided. What do you think of setting logger to a null logger instead @sergiyv-bitquill @ColinKYuen

Reply:
I'd be against setting logger to null. That add complexity to the code like checking if logger is null or not. It's better to make sure to pass a dummy logger.

Reply:
I was referring to the class `com.mysql.cj.log.NullLogger`. I think we are on the same page here.